library/chemistry/src/Tests.qs

```diff
@@ -365,3 +404,8 @@ function NearEqualityFactD(actual : Double, expected : Double) : Unit {
         fail $"Values were not equal within tolerance\nActual: {actual}, Expected: {expected}, Tolerance: {tolerance}";
     }
 }
+
+// For convenience of constructing complex constants
+function NewComplex(r : Double, i : Double) : Complex {
```

Comment:
This feels slightly redundant given the existence of the function call constructor, but that's also what we are trying to avoid... looking at the places where this is used, all except two are part of constructing a `JordanWignerInputState` so maybe a way to make this helper seem more helpful is to have it be a convenience method for constructing those. Then the remaining two spots can use the normal `new Complex` style.

Reply:
We had a short discussion here in the office about this way of contructing struct and the lint we have about it. When it is a big struct, and is not constructed very often, this field-by field assignments are perfect. But when it is a simple struct and you need to construct a bunch of literals, spelling it all out is kind-of annoying (I tried!). And I'm talking specifically about constants. It is rather natural and anambiguous to write something like Complex(1.0, 0.0).
Could we extend Q# to support imginary unit literal? I want to write 1.0+0.0i :)

Reply:
I moved construction to NewJordanWignerInputState instead, but it feels like pushing the same problem to a different place. We'll discuss later.